tests/test_client.py

```diff
@@ -286,3 +292,67 @@ def test_batch_wait(self, client):
         for pos, success in success_tasks:
             assert(tasks[pos].pk == success.pk)
             assert inference_schema(2, 0, 'golden retriever', 0.8) == success['data']
+
+
+class TestClientRetry(object):
+    DEFAULT_TIMEOUT = 5
+    DEFAULT_MIN_ATTEMPT_NUMBER = 4
+
+    def expect_retry(self, client, timeout, min_attempt_number):
```

Comment:
from the name I was expecting it would just check, not do a request too, either explicit name or move the retrieve outside